test/library_search/basic.lean

```diff
@@ -70,9 +70,50 @@ by library_search -- says: `exact add_pos ha hb`
 example (a b : ℕ) : 0 < a → 0 < b → 0 < a + b :=
 by library_search -- says: `exact add_pos`
 
+section synonym
+
+-- Synonym `>` for `<` in the goal
+example (a b : ℕ) : 0 < a → 0 < b → a + b > 0 :=
+by library_search -- says: `exact add_pos`
+
+-- Synonym `>` for `<` in another part of the goal
+example (a b : ℕ) : a > 0 → 0 < b → 0 < a + b :=
+by library_search -- says: `exact add_pos`
+
+-- Synonym `>` for `<` in another part of the goal
+example (a b : ℕ) (ha : a > 0) (hb : 0 < b) : 0 < a + b :=
+by library_search -- says: `exact add_pos ha hb`
+
 example (a b : ℕ) (h : a ∣ b) (w : b > 0) : a ≤ b :=
 by library_search -- says: `exact nat.le_of_dvd w h`
 
+example (a b : ℕ) (h : a ∣ b) (w : b > 0) : b ≥ a :=
+by library_search -- says: `exact nat.le_of_dvd w h`
+
+-- A lemma with head symbol `¬` can be used to prove `¬ p` or `⊥`
+example (a : ℕ) : ¬ (a < 0) := by library_search -- says `exact not_lt_bot`
+
+example (a : ℕ) (h : a < 0) : false := by library_search -- says `exact not_lt_bot h`
+
+def P : ℕ → Prop := λ _, sorry
```

Comment:
Is it possible to arrange these tests in a way that doesn't use `sorry`? I don't have a concrete objection, just the feeling that we should avoid `sorry` in mathlib entirely.

Reply:
OK! I wanted to ensure that the test cases don't get solved "accidentally" without using the lemmas that we want to test (e.g. by the corresponding lemma being added to mathlib). It looks like introducing a new inductive type also does the trick.